### ISSUE TITLE
job-manager: add secondary priority value

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#3256]
Following up on meeting discussion, job manager needs to track a secondary job priority value that can be updated by flux-accounting components. We need to do some design on an interface for updating this priority.

Let's start by finding related issues and linking them in here.

flux-framework/flux-accounting#7 has some architecture discussion
See also #1708

**Comment by grondo** [comment]
Some possible related components were mentioned in flux-framework/flux-accounting/issues/7

**Comment by garlick** [comment]
I think our design, discussed in #3311 and elsewhere, is evolving towards one where the job manager's alloc queue is ordered by a single priority value, calculated by plugin, and taking into account t_submit and the administrative priority (possibly among other things).

RFC 21 has a PR pending that would define this new priority as a double precision floating point value between 0 and 1.  To complete this issue, we could add that value to the job manager's `struct job`  and calculate it based on the above two values such that the queue order is the same as it is now.   (And recalculate it when the administrative priority is updated).  That work should not begin until the RFC 21 changes are merged though.

Another issue can cover implementing the default priority function as a priority plugin.

**Comment by garlick** [comment]
I think this is resolved now that the job manager knows about priority.
